### ISSUE TITLE
[Feature Request] Gamma transitioning

### ISSUE THREAD

**Issue by silvercomet** [anthonya1999/GoodNight#78]
So first of all, I would like to thank you, the devs of this app, all for making f.lux feature available for all of us at a better comfortability after Thomas Finch seems liek has abandoned the gamma thingy project for whatever reason I don't know.

Goodnight has been a great help to me and I like how it integrates well with background app refresh and is able to change the gamma almost immediately, and doesn't require the notification to wake screen to turn on like the sideloaded flux version 1. 

SInce it's so great like that, I would like to suggest, why not make it even more like f.lux on windows. If either of you have followed the project done by Thomas, then later forked by Amangelo, he worked on a transition branch for the app, so the color will adjust gradually for 1h or 30 min beforehand to let the eye adapt better to the change of color. This I think is extremely helpful for Goodnight when changing from Night mode to bedtime mode, which is a lot of increase of orangeness there. 

In application-wise, I would suggest that the iOS framework kit will just divide the gamma value evenly maybe by 10 minute block within a set timeframe of transition. So option of 1 hour transition beforehand the orangeness will adjust 6 times during the period. It will be even better if then we could choose the number of hour of transition in settings as well as transition blocks. That might runs a bit of battery there but I think it's totally worth it, for the better health.

Thank you guys for making an incredibly helpful app

**Comment by mariokorte** [comment]
The location based switching already does that...

**Comment by silvercomet** [comment]
Oh you mean its already implemented in the location based switch? I never used the location based switch before, but does it requires internet connection to work? Because I dont always leave cellular data on so when I go out, basically there is no wifi or 3g. Or does it just use the cellular network to determine the location and will auto adjust to it?

**Comment by silvercomet** [comment]
Since I leave goodnight running in the background all the time, it will also use the location service all the time but does that affect battery much comparing to the method of time changing?

**Comment by silvercomet** [comment]
I might have not put it clearly but I just want to make it clear that I was talking about the toggle option of flux in windows that let you decide whether the change will take effect immediately or gradually so the eyes adapt better to that. Its a separated option in addition to using location based switch. I was just not sure if the location based switch of goodnight does that before since I didnt see it bring documented. I just feel that this app is so good and having this one will be another added value to the perfection of the app and totally makes it flux on pc or mac on the mobile

**Comment by shad0whawk** [comment]
Since we are on the topic already, I'd like to know what the trigger is for automatically changing the temperature while using the location option. Does it key off of the sunset/sunrise time or something else?  Does the change happen instantly or does it incrementally decrease/increase over a period of time?

**Comment by mariokorte** [comment]
Yes, it's based upon sunset and sunrise at your location and it fades slowly to the desired color. Just try it for yourself.

**Comment by mariokorte** [comment]
@silvercomet it doesn't need a data connection and the location service uses the setting with the lowest battery usage. This nearly has no impact at all on your battery...

**Comment by shad0whawk** [comment]
Will the widget reflect the incremental changes if I look at it while it's in the transition period?

**Comment by silvercomet** [comment]
@mariokorte thank you, i am trying it out today, will let you know the result

**Comment by silvercomet** [comment]
But does the fading effect also apply when changing from night mode to bedtimw mode though?

**Comment by mariokorte** [comment]
No, currently not.

**Comment by silvercomet** [comment]
Ok I just want to confirm that. Switching from time based to location based does a wonderful job of creating a fading transition effect that is well adaptive and pleasing to the eyes. I hope that you guys could implement this fading transition for the switch between night mode and bedtime mode since that is still a very large gap of gamma to change in a sudden. I also feel like this transitioning effect should be documented in the Readme file since its different from the way flux work a bit unless you turn on the option of having effect to take effect gradually. Just my thought on this since i don't know if people all know about this
